xonsh/ptk/shell.py

```diff
@@ -39,6 +39,9 @@ def __init__(self, **kwargs):
         self.key_bindings_manager = KeyBindingManager(**key_bindings_manager_args)
         load_xonsh_bindings(self.key_bindings_manager)
 
+        self._prompt_tokens = None
```

Comment:
Hey @Carreau -- re the `_prompt_tokens` stuff, I agree with the changes but is it part of this feature?